### PR TITLE
Binary mera

### PR DESCRIPTION
**Pull request by mganahl**
This PR adds an optimization of a scale-invariant binary MERA network to approximate
the ground-state of the transverse field Ising model at criticality.

it includes:
1. benchmarks for ascending and descending operations
2. benchmark for optimizations
3. code for computing scaling dimensions

Additionally, almost all contractions are carried out by directly using the TensorNetwork API
(instead of the ncon wrapper).

### CODE REVIEW

**Comment by Thenerdstation** [comment]
Let me know when you are ready to merge this.

**Comment by mganahl** [comment]
Hey @Thenerdstation, I think this can be merged

**Comment by Thenerdstation** [comment]
looks good afaict. Can you add a small unit test? That will ensure this code doesn't die when we do inevitable changes.

**Comment by mganahl** [comment]
sure, what's the preferred way of testing (i.e. unittest, pytest, tensorflow test suite)?

**Comment by Thenerdstation** [comment]
We use pytest, but pytest can also run all 3. do what you think is easiest.

File: experiments/MERA/test_binary_mera.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# Copyright 2019 The TensorNetwork Authors
```
Comment (Thenerdstation) [inline experiments/MERA/test_binary_mera.py]: change to binary_mera_test.py for consistency.

Comment (mganahl) [reply experiments/MERA/test_binary_mera.py]: renamed and added a few more tests.

**Pull request merged**